scipy/linalg/tests/test_lapack.py

```diff
@@ -559,3 +559,43 @@ def test_sgesdd_lwork_bug_workaround():
     assert_equal(returncode, 0,
                  "Code apparently failed: " + p.stdout.read())
 
+
+class TestSytrd(object):
+    def test_sytrd(self):
+        np.random.seed(42)
+        n = 3
+        A = np.random.normal(size=(n, n))
+        A += A.T
+
+        sytrd, sytrd_lwork = get_lapack_funcs(('sytrd', 'sytrd_lwork'))
```

Comment:
You would add the array you want to get the LAPACK routine for as `get_lapack_funcs(('sytrd', 'sytrd_lwork'), (A,))` such that `d` is selected. Same should also test for `s` flavor with a `float32` array.